temporal-sdk/src/main/java/io/temporal/common/converter/SearchAttributesPayloadConverter.java

```diff
@@ -0,0 +1,52 @@
+/*
+ *  Copyright (C) 2020 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.common.converter;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.protobuf.ByteString;
+import io.temporal.api.common.v1.Payload;
+import io.temporal.internal.common.SearchAttributesUtil;
+import java.util.Optional;
+
+public class SearchAttributesPayloadConverter extends JacksonJsonPayloadConverter {
+
+  public SearchAttributesPayloadConverter() {
+    super();
+  }
+
+  @Override
+  public Optional<Payload> toData(Object value) throws DataConverterException {
+    try {
+      byte[] serialized = mapper.writeValueAsBytes(value);
+      return Optional.of(
+          Payload.newBuilder()
+              .putMetadata(EncodingKeys.METADATA_ENCODING_KEY, EncodingKeys.METADATA_ENCODING_JSON)
+              .putMetadata(
+                  EncodingKeys.METADATA_TYPE_KEY,
```

Comment:
Why do we store this field here, but don't use it for deserialization and instead of this use this field in the Util class?